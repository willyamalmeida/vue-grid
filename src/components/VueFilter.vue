<template>
        <form class="navbar-form navbar-left">
            <div class="form-group">
                <input
                    type="text"
                    class="form-control form-control-sm"
                    placeholder="Search"
                    maxlength="100"
                    :value="value"
                    @input="updateValue($event.target.value)" />
            </div>
        </form>
</template>

<script>
export default {
    name: "vue-filter",
    props: ["value"],

    methods: {
        updateValue(value) {
            this.$emit("input", value);
        }
    }
}
</script>


