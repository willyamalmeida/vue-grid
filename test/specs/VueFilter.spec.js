import { mount } from "vue-test-utils";
import VueFilter from "./../../src/components/VueFilter.vue";

const createVueFiler = function(propValue){
    return mount(VueFilter, {
        propsData: {
            value: propValue
        }
    });
}

describe("VueFilter", () => {
    it("test has a methods hook", () => {
        expect(typeof VueFilter.methods).toBe("object");
    });

    it("test name of component", () => {
        expect(VueFilter.name).toBe("vue-filter");
    });

    it("test is a Vue component", () => {
        const vueFilter = mount(VueFilter);
        expect(vueFilter.isVueInstance()).toBeTruthy();
    });

    it("test props value", () => {
        const wrapper = createVueFiler("my filter");
        let value = wrapper.props().value;

        expect(value).toBe("my filter");
    });


    it("test update input value", () => {
        const expected = "new filter";
        const propValue = "my filter";
        const wrapper = createVueFiler(propValue);

        let input = wrapper.find("input");

        expect(input.element.value).toBe(propValue);

        input.element.value = expected;
        input.trigger("input");

        let emitted = wrapper.emitted("input");

        expect(emitted).toBeTruthy();
        expect(emitted.length).toBe(1);

        //emitted[0] = ["value"]
        //emitted[0][0] = "value"
        let value = emitted[0][0];

        expect(value).toBe(expected);
    });

    it("test VueFilter html structure", () => {
        const wrapper = createVueFiler("my filter");

        expect(wrapper.element).toMatchSnapshot();
    });
});
