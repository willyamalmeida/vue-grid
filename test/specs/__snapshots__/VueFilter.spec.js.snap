// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VueFilter test VueFilter html structure 1`] = `
<form
  class="navbar-form navbar-left"
>
  <div
    class="form-group"
  >
    <input
      class="form-control form-control-sm"
      maxlength="100"
      placeholder="Search"
      type="text"
    />
  </div>
</form>
`;
